libs/pulumi-static-site/src/cloudfront.ts

```diff
@@ -0,0 +1,123 @@
+import * as aws from '@pulumi/aws'
+import * as pulumi from '@pulumi/pulumi'
+import * as random from '@pulumi/random'
+
+export class Distribution extends pulumi.ComponentResource {
+    distribution: aws.cloudfront.Distribution
+
+    constructor(
+        name: string,
+        args: {
+            acmCertificateArn?: pulumi.Input<string>
+            domains: pulumi.Input<string>[] | pulumi.Input<string[]>
+            originDomainName: pulumi.Input<string>
+            priceClass?: pulumi.Input<string>
+            cachePolicyId?: pulumi.Input<string>
+            originRequestPolicyId?: pulumi.Input<string>
+        },
+        opts?: pulumi.ComponentResourceOptions,
+    ) {
+        super(
+            'swm:pulumi-static-site:distribution/Distribution',
+            name,
+            {},
+            opts,
+        )
+
+        const originId = new random.RandomString(
+            `${name}-originId`,
+            {
+                length: 8,
+                special: false,
+                number: false,
+            },
+            { parent: this },
+        )
+
+        const cachePolicyId = pulumi
+            .all([
+                args.cachePolicyId,
+                pulumi.output(
+                    aws.cloudfront.getCachePolicy(
```

Comment:
Why do you need to get the cache policy? The guid is hard coded and just using a constant with a JSDoc does the same thing without the additional complexity and call?

Reply:
Didn't realise it was a constant, sounds good.